lenskit-core/src/main/java/org/lenskit/data/entities/BuiltBy.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2014 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.lenskit.data.entities;
+
+import org.lenskit.data.events.EventBuilder;
+
+import java.lang.annotation.*;
+
+/**
+ * Identify the builder type for an entity class.
+ */
+@Documented
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface BuiltBy {
```

Comment:
Yes, for code that isn't in this PR. The entity builders are used by file loading code and by the view type transmogrification logic.

You'll be able to tell the DAO 'give me the `rating` entities, and I want to see them as type `Rating`', and if the objects aren't already `Rating` objects, it will look at `Rating`'s `@BuiltBy` to figure out the builder class to use, and use that to convert the object to a `Rating`.